cmd/maya-apiserver/app/server/restore_endpoint_v1alpha1.go

```diff
@@ -255,3 +277,39 @@ func updateRestoreStatus(clientset versioned.Interface, rst v1alpha1.CStorRestor
 		return
 	}
 }
+
+func createVolumeForRestore(r *v1alpha1.CStorRestore) (*v1alpha1.CASVolume, error) {
+	vol := &v1alpha1.CASVolume{}
+
+	vol.Name = r.Spec.VolumeName
+	vol.Labels = map[string]string{
+		string(v1alpha1.StorageClassKey): r.Spec.StorageClass,
+	}
+	vol.Spec.Capacity = r.Spec.Size.String()
+
+	if r.Spec.Local {
+		vol.CloneSpec.IsClone = true
+		vol.CloneSpec.SourceVolume = r.Spec.RestoreSrc
+		vol.CloneSpec.SnapshotName = r.Spec.RestoreName
+	} else {
+		vol.Annotations = map[string]string{
+			v1alpha1.PVCreatedByKey: "restore",
+		}
+	}
+
+	vOps, err := volume.NewOperation(vol)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create volume operation")
+	}
+
+	cvol, err := vOps.Read()
+	if err != nil {
+		if !isNotFound(err) {
+			return nil, errors.Wrapf(err, "Failed to get restore volume details")
+		}
+	} else {
+		return cvol, nil
+	}
+
+	return vOps.Create()
```

Comment:
is this supposed to be for remote backup?

Reply:
It serves for local and cloud backup. As of now we are creating volume through PVC from velero-plugin in case of remote restore So this will not execute for remote backup. https://github.com/openebs/maya/pull/1575#discussion_r398099833